folioreader/src/main/java/com/folioreader/ui/base/SaveReceivedHighlightTask.java

```diff
@@ -0,0 +1,43 @@
+package com.folioreader.ui.base;
+
+import android.os.AsyncTask;
+
+import com.folioreader.model.HighlightImpl;
+import com.folioreader.model.sqlite.HighLightTable;
+
+import java.util.List;
+
+/**
+ * Created by gautam on 10/10/17.
+ */
+
+public class SaveReceivedHighlightTask extends AsyncTask<Void, Void, Void> {
+
+    private OnSaveHighlight onSaveHighlight;
+    private List<HighlightImpl> highlights;
+    private String bookId;
+
+    public SaveReceivedHighlightTask(OnSaveHighlight onSaveHighlight,
+                                     List<HighlightImpl> highlights,
+                                     String bookId) {
```

Comment:
it doesn't need bookId.. If user is providing Highlight, it must provide bookId..